#22885

kgelinas opened the pull request:
#22885 fixing hardcoded value in vsphere init.sh

<!-- Reviewable:start -->

---

This change is [<img src="http://reviewable.k8s.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/24708)

<!-- Reviewable:end -->

jsafrane committed:
e2e test for dynamic provisioning.

The test works only on GCE, AWS and OpenStack. It creates and deletes a 2GiB
volume.

freehan committed:
fixing dead endpoint black hole udp traffic

spxtr committed:
Use SCP to dump logs and parallelize a bit.

Q-Lee committed:
Support addon Deployments, make heapster a deployment with a nanny.

Commit:
Merge pull request #22942 from jsafrane/devel/aws-more-devices

Allow up to 39 PVs attached to AWS node
(cherry picked from commit 8a952748a1e0462faa9304cdcdd5a215614b40aa)

Commit:
Merge pull request #22969 from nikhiljindal/upgradeTest

Auto commit by PR queue bot
(cherry picked from commit 52b061a4d3b48fd47528b83e60fd76b29aecf1d5)

Commit:
Merge pull request #22993 from piosz/bump-heapster-1.0

Bumped Heapster to stable version 1.0.0
(cherry picked from commit 70036b3621dabc5ff60cb02e637f3c03f7583088)

Commit:
Merge pull request #22948 from janetkuo/e2e-test-not-mutate

Auto commit by PR queue bot
(cherry picked from commit 4b3e2b57ac4e944da1d7b0d3d73a03e0e34f7032)

Commit:
Merge pull request #23012 from caesarxuchao/fix-22996

Make fullyLabeledReplicas optional in rc status
(cherry picked from commit 6170326db11acd008d89b90263d2ae49d7b51ba6)

Commit:
Merge pull request #23005 from andyzheng0831/fix

Auto commit by PR queue bot
(cherry picked from commit 83135047285555ca92df30075f6fb01e22c1160c)

Commit:
Merge pull request #23020 from vishh/godeps-cadvisor

Updating cadvisor deps to v0.22.2
(cherry picked from commit 57bd9d4289a2edbd0da9d896dc0fe236a3be77a5)

bgrant0607 committed:
Merge pull request #23024 from eparis/release-1.2

Release 1.2 cherrypick batch

Commit:
Merge pull request #22921 from liggitt/job-validation

Auto commit by PR queue bot
(cherry picked from commit 17d8e612442b189024c3ab69cb0f92710c398318)

mwielgus committed:
Merge pull request #23037 from piosz/addons-doc

Updated documentation about monitoring addons
(cherry picked from commit 30bc3e579e49cfb50f014a7f0b955bc62a26af40)

tobad357 committed:
Fix for issue #22932 infinite pod restarts

This fixes an issue when using CNI where the hash of a Container object will differ between creation and change checks due to the docker image exporting ports

bgrant0607 committed:
Merge pull request #23051 from eparis/release-1.2

Release 1.2 cherrypick batch 16/3 morning

bgrant0607 committed:
Merge pull request #23058 from eparis/automated-cherry-pick-of-#23050-upstream-release-1.2

Automated cherry pick of #23050

david-mcmahon committed:
Kubernetes version v1.2.0

david-mcmahon committed:
Kubernetes version v1.2.1-beta.0

Q-Lee committed:
Removing versions from deployment names.

Commit:
Merge pull request #22960 from andyzheng0831/trusty

Auto commit by PR queue bot
(cherry picked from commit 5cc2bb3c0c9318ac6168351252b50f1dba71a8c9)

Commit:
Merge pull request #22974 from mesosphere/jdef-forward-housekeeping-flags

Auto commit by PR queue bot
(cherry picked from commit 68892a3e59c63fa7de2582d20642498e8cab4132)

Commit:
Merge pull request #23035 from xinxiaogang/xnxin-master

Auto commit by PR queue bot
(cherry picked from commit cda4583984ada38e61aa1b907aae357c4a80a467)

Q-Lee committed:
Use the same constants as heapster for the heapster nanny.

Q-Lee committed:
Adding e2e test support for monitoring deployments.

Commit:
Merge pull request #23143 from vishh/23113

Auto commit by PR queue bot
(cherry picked from commit 78adb885c41dcbb820b6e79c975e3ddb91ad5700)

Commit:
Merge pull request #23167 from cjcullen/sshcheck

Auto commit by PR queue bot
(cherry picked from commit 10204f8b31bdc0ab2accbb70cdca304df2231c82)

Commit:
Merge pull request #23141 from yujuhong/fix_race

Auto commit by PR queue bot
(cherry picked from commit 61b9a21cfd7bca8fbedff875647d7b7e0aff5603)

Commit:
Merge pull request #23145 from liggitt/patch-internal-version

Auto commit by PR queue bot
(cherry picked from commit 62399077d86cc116c8cc4570c399da726bcfb48d)

mikedanese committed:
don't sync daemonsets or controllers with selectors that match all pods

Commit:
Merge pull request #23222 from madhusudancs/kubectl-expose-deployment-example

Auto commit by PR queue bot
(cherry picked from commit 2c5903acf1f814d5f4cb39056b75d82179b2d08f)

Commit:
Merge pull request #23019 from alex-mohr/oplimit

Auto commit by PR queue bot
(cherry picked from commit 0fe049f9ffb82b94e850270ccfa710ee3ed3ebd3)

Commit:
Merge pull request #23246 from AdoHe/update_expose_example

Auto commit by PR queue bot
(cherry picked from commit b077b685d97842449ecddaecfff9b67baa150bf8)

Commit:
Merge pull request #23079 from andyzheng0831/hybrid

Auto commit by PR queue bot
(cherry picked from commit d5316c21ef0603cc5e858284e7786e7647cb307f)

Commit:
Merge pull request #23160 from janetkuo/rollback-copy-annotation

Auto commit by PR queue bot
(cherry picked from commit edc35bfebaeecd6599110c490c1fb1f33289f209)

Commit:
Merge pull request #23180 from roberthbailey/dead-code

Auto commit by PR queue bot
(cherry picked from commit f2d14990a51b7915ed0d15f65a5803047d304b9e)

Commit:
Merge pull request #23313 from zmerlynn/container-vm-v20160321

Auto commit by PR queue bot
(cherry picked from commit 4d98abf26c38e5f77675fcd5821c104a3e5d0580)

justinsb committed:
AWS: Fix problems with >2 security groups

The previous logic was incorrect; if we saw two untagged security groups
before seeing the first tagged security, we would incorrectly return an
error.

Fix #23339

Commit:
Merge pull request #23325 from MurgaNikolay/master

Auto commit by PR queue bot
(cherry picked from commit 8f71655bcc3f06af162f16e8f87488fab438609b)

Commit:
Merge pull request #23234 from nikhiljindal/swaggerRemoveurlQuery

Auto commit by PR queue bot
(cherry picked from commit b50e89e323977d365439fda42d75f0930310a517)

Commit:
Merge pull request #23305 from janetkuo/update-kubectl-help

Auto commit by PR queue bot
(cherry picked from commit 2f8986782178900425df02e32338981b6602d8ec)

janetkuo committed:
Update guestbook examples; replace RC concepts with Deployment

Commit:
Merge pull request #22818 from andyzheng0831/enhance

Auto commit by PR queue bot
(cherry picked from commit f8bb10b479bf6f6c3f8db5213edab766bc5fa873)

Commit:
Trusty: Update heapster manifest handling code

jsafrane committed:
Fixed mounting with containerized kubelet.

Kubelet was not able to mount volumes when running inside a container and
using nsenter mounter,

NsenterMounter.IsLikelyNotMountPoint() should return ErrNotExist when the
checked directory does not exists as the regular mounted does this and
some volume plugins depend on this behavior.

Commit:
Support differentiation of OS distro in e2e tests

Commit:
Update port forward e2e for go 1.6

Only close the stdout/stderr pipes from kubectl port-forward when we're truly done with the command,
instead of as soon as runPortForward exits.

Also try to gracefully stop kubectl port-forward via SIGINT, instead of always sending SIGKILL, as
this will help avoid spdy goroutine leaks in the Kubelet.

vishh committed:
Do not track resource usage for host path volumes. They can contain loops.

Signed-off-by: Vishnu kannan <vishnuk@google.com>

bgrant0607 committed:
Merge pull request #23450 from eparis/release-121

Cherry-picks for release 1.2 - 24 Mar

mikedanese committed:
add an event for when a daemonset can't place a pod due to insufficent resource or port conflict

Commit:
Merge pull request #23457 from vishh/automated-cherry-pick-of-#23446-upstream-release-1.2

Auto commit by PR queue bot

mikedanese committed:
don't sync deployment when pod selector is empty

piosz committed:
Heapster patch release to 1.0.2

Q-Lee committed:
Create a new Deployment in kube-system for every version.

Commit:
Merge pull request #23456 from mikedanese/automated-cherry-pick-of-#23223-upstream-release-1.2

Auto commit by PR queue bot

mikedanese committed:
validate that daemonsets don't have empty selectors on creation

Commit:
Trusty: Regional release .tar.gz support

david-mcmahon committed:
Honor incoming OUT_DIR from the Makefile.

tam7t committed:
IngressTLS: allow secretName to be blank for SNI routing

dchen1107 committed:
Remove unnecessary override of /etc/init.d/docker on containervm image.

bgrant0607 committed:
Merge pull request #23543 from andyzheng0831/automated-cherry-pick-of-#23466-upstream-release-1.2

Automated cherry pick of #23466

bgrant0607 committed:
Merge pull request #23586 from mikedanese/automated-cherry-pick-of-#23530-#23467-upstream-release-1.2

Automated cherry pick of #23530 #23467

bgrant0607 committed:
Merge pull request #23617 from david-mcmahon/automated-cherry-pick-of-#23561-upstream-release-1.2

Automated cherry pick of #23561

cjcullen committed:
Change kube-proxy & fluentd CPU request to 20m/80m.

Random-Liu committed:
Fix TerminationMessagePath

ArtfulCoder committed:
make docker-checker more robust

maclof committed:
Drain pods created from ReplicaSets

bgrant0607 committed:
Merge pull request #23687 from ArtfulCoder/automated-cherry-pick-of-#23662-upstream-release-1.2

Automated cherry pick of #23662

bgrant0607 committed:
Merge pull request #23685 from cjcullen/automated-cherry-pick-of-#23646-upstream-release-1.2

Automated cherry pick of #23646

bgrant0607 committed:
Merge pull request #23701 from dchen1107/automated-cherry-pick-of-#23593-upstream-release-1.2

Remove unnecessary override of /etc/init.d/docker on containervm image.

bgrant0607 committed:
Merge pull request #23487 from piosz/heapster-1.0.1

Heapster patch release to 1.0.2

bgrant0607 committed:
Merge pull request #23707 from bprashanth/automated-cherry-pick-of-#23500-upstream-release-1.2

Automated cherry pick of #23500

Commit:
Update Dashboard UI addon to v1.0.1 Release details: https://github.com/kubernetes/dashboard/releases/tag/v1.0.1

Commit:
Merge pull request #23718 from justinsb/automated-cherry-pick-of-#23340-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #23340

Cherry pick of #23340 on release-1.2.

derekwaynecarr committed:
A pod never terminated if a container image registry was unavailable

david-mcmahon committed:
Kubernetes version v1.2.1

david-mcmahon committed:
Kubernetes version v1.2.2-beta.0

luxas committed:
Fix so setup-files don't recreate/invalidate certificates that already exist

Commit:
add the rm in verify-api-reference-docs.sh back

saad-ali committed:
Ensure volume GetCloudProvider code uses cloud config

justinsb committed:
AWS: Don't error if there are no ephemeral disks

format-disks used to run with non-strict bash semantics, but this changed in
1.2 as we now merge it into the GCE script, so pipefail and errexit are both
set.

However, the way we list the ephemeral disks, by piping to grep, would cause an
exit code of 2 if there were no ephemeral disks.

Tolerate failure here by add `|| true`.  The metadata service call is unlikely
to fail, so we continue to ignore that possibility.

justinsb committed:
AWS kube-up: Allow BLOCK_DEVICE_MAPPINGS_BASE to be empty

We rename it to EPHEMERAL_BLOCK_DEVICE_MAPPINGS, and we also change the value
so that it starts with a `,`, instead of always inserting a comma before it.
In this way the value can be empty.

Also, if the user sets the (currently experimental) KUBE_AWS_STORAGE
environment variable to be "ebs", then we will not mount any instance storage
which will cause the machines to use EBS storage instead.

pmorie committed:
Make ConfigMap volume readable as non-root

caesarxuchao committed:
Merge pull request #23755 from caesarxuchao/release-1.2-add-rm-back

Add the rm in verify-api-reference-docs.sh back

cjcullen committed:
Add a timeout to the sshDialer to prevent indefinite hangs.

vishh committed:
OOM Score adjusting logic in kubelet should handle containers that exit quickly.

Signed-off-by: Vishnu kannan <vishnuk@google.com>

mikedanese committed:
only include running and pending pods in daemonset should place calculation

zmerlynn committed:
Merge pull request #23783 from luxas/automated-cherry-pick-of-#23550-upstream-release-1.2

Automated cherry pick of #23550 upstream release 1.2

zmerlynn committed:
Merge pull request #23938 from zmerlynn/automated-cherry-pick-of-#23776-upstream-release-1.2

Automated cherry pick of #23776

zmerlynn committed:
Merge pull request #23901 from cjcullen/automated-cherry-pick-of-#23843-upstream-release-1.2

Automated cherry pick of #23843

janetkuo committed:
Don't upgrade heapster version (keep v1.0.2)

zmerlynn committed:
Merge pull request #23943 from saad-ali/automated-cherry-pick-of-#21140-#23769-upstream-release-1.2

Automated cherry pick of #21140 #23769 upstream release 1.2

zmerlynn committed:
Merge pull request #23937 from zmerlynn/automated-cherry-pick-of-#23724-upstream-release-1.2

Automated cherry pick of #23724

zmerlynn committed:
Merge pull request #23824 from derekwaynecarr/automated-cherry-pick-of-#23746-upstream-release-1.2

Automated cherry pick of #23746 upstream release 1.2

zmerlynn committed:
Merge pull request #23934 from mikedanese/automated-cherry-pick-of-#23929-#23463-upstream-release-1.2

Automated cherry pick of #23929 #23463

zmerlynn committed:
Merge pull request #23935 from mikedanese/automated-cherry-pick-of-#23793-upstream-release-1.2

Automated cherry pick of #23793

zmerlynn committed:
Merge pull request #23836 from andyzheng0831/automated-cherry-pick-of-#23558-upstream-release-1.2

Automated cherry pick of #23558

zmerlynn committed:
Merge pull request #23951 from spxtr/automated-cherry-pick-of-#22835-upstream-release-1.2

Use SCP to dump logs and parallelize a bit.

zmerlynn committed:
Merge pull request #23941 from janetkuo/automated-cherry-pick-of-#22893-#23512-upstream-release-1.2

Automated cherry pick of #22893 #23512

vishh committed:
Adding an e2e test for oom score adjust handling.

Signed-off-by: Vishnu kannan <vishnuk@google.com>

Commit:
Merge pull request #23965 from janetkuo/automated-cherry-pick-of-#23408-upstream-release-1.2

Automated cherry pick of #23408

tomdee committed:
handle empty perferredVersion

Commit:
Merge pull request #23986 from zmerlynn/automated-cherry-pick-of-#23434-upstream-release-1.2

Automated cherry pick of #23434

Commit:
Trusty: Do not create the docker-daemon cgroup

yujuhong committed:
e2e: adapt kubelet_perf.go to use the new summary metrics API

This commit switch most functions in kubelet_stats.go to use the new API.
However, the functions that perform one-time resource usage retrieval remain
unchanged to be compatible with reource_usage_gatherer.go. They should be
handled separately.

Also, the new summary API does not provide the RSS memory yet, so all memory
checking tests will *always* pass. We plan to add this metrics in the API and
restore the functionality of the test.

Commit:
make default thirdparty resource preferredVersion; adding back discovery tests

yujuhong committed:
kubele: add RSS memroy to the summary API

liggitt committed:
Make watch cache behave like uncached watch

david-mcmahon committed:
Kubernetes version v1.2.2

david-mcmahon committed:
Kubernetes version v1.2.3-beta.0

jsafrane committed:
Add path to log messages.

It is useful to see the path that failed to mount in the log messages.

nikhiljindal committed:
Allowing type object in kubectl swagger validation

j3ffml committed:
Fix list-resources grep

Commit:
Merge pull request #24063 from jlowdermilk/automated-cherry-pick-of-#24058-upstream-release-1.2

Automated cherry pick of #24058

Commit:
Trusty: Avoid unnecessary in-memory temp files

Commit:
Add ClusterUpgrade tests

Commit:
Trusty: Fix master kubelet command assembly

freehan committed:
add test for udp connection flush

yujuhong committed:
e2e: fix error checking in kubelet stats

Commit:
Honor starting resourceVersion in watch cache

Compare the requested resourceVersion to each event's resourceVersion to ensure events that occurred
in the past are not sent to the client.

Commit:
Merge pull request #24060 from andyzheng0831/automated-cherry-pick-of-#23996-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #23996

Cherry pick of #23996 on release-1.2.

lavalamp committed:
Merge pull request #24150 from ihmccreery/split-upgrade-jobs-1.2

Add ClusterUpgrade tests

Commit:
Merge pull request #24104 from nikhiljindal/automated-cherry-pick-of-#24054-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24054

Cherry pick of #24054 on release-1.2.

Commit:
Merge pull request #24219 from andyzheng0831/automated-cherry-pick-of-#24144-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24144

Cherry pick of #24144 on release-1.2.

lavalamp committed:
Merge pull request #24135 from liggitt/automated-cherry-pick-of-#24008-upstream-release-1.2

Automated cherry pick of #24008

lavalamp committed:
Merge pull request #24258 from andyzheng0831/automated-cherry-pick-of-#24172-upstream-release-1.2

Automated cherry pick of #24172

errm committed:
Allow setting the Host header in a httpGet probe

Fixes #24288

Commit:
Version-guard Kubectl client Guestbook application test against deployments

Commit:
Trusty: Add debug supports for docker and kubelet

Commit:
Add configuration for splitting upgrade jobs instead of using Jenkins steps

cjcullen committed:
Masquerade traffic from off-cluster going through kube-proxy.

yujuhong committed:
Remove host port in influxdb-grafana-controller.yaml

There is no valid reason to keep the host ports in the pod spec.

a-robinson committed:
Fix gce.getDiskByNameUnknownZone logic.

Commit:
Merge pull request #24227 from yujuhong/automated-cherry-pick-of-#24015-#24003-#24205-upstream-release-1.2

Automated cherry pick of #24015 #24003 #24205

cjcullen committed:
Fix goroutine leak in ssh-tunnel healthcheck.

a-robinson committed:
Merge pull request #24491 from a-robinson/automated-cherry-pick-of-#24452-upstream-release-1.2

Automated cherry pick of #24452 upstream release 1.2

Commit:
Trusty: Handle the new var in kube-proxy manifest

zmerlynn committed:
Merge pull request #24504 from Random-Liu/automated-cherry-pick-of-#23658-upstream-release-1.2

Automated cherry pick of #23658

zmerlynn committed:
Merge pull request #24307 from ncdc/automated-cherry-pick-of-#24208-upstream-release-1.2

Automated cherry pick of #24208

zmerlynn committed:
Merge pull request #24485 from vishh/automated-cherry-pick-of-#23894-upstream-release-1.2

Automated cherry pick of #23894 upstream release 1.2

zmerlynn committed:
Merge pull request #24244 from jsafrane/automated-cherry-pick-of-#23435-upstream-release-1.2

Automated cherry pick of #23435

zmerlynn committed:
Merge pull request #24497 from cjcullen/automated-cherry-pick-of-#24487-upstream-release-1.2

Automated cherry pick of #24487

zmerlynn committed:
Merge pull request #24478 from ihmccreery/cherry-pick-24342-1.2

Automated cherry pick of #24342

Commit:
Merge pull request #24428 from ihmccreery/report-prefix-1.2

Automatic merge from submit-queue

Add configuration for splitting upgrade jobs instead of using Jenkins steps

This is a manual cherrypick of another part of #22962 (different from #24150).  This properly propagates `E2E_REPORT_PREFIX`, since the files changed here are versioned differently from `e2e-runner.go`.

cc @roberthbailey @zmerlynn as owners of v1.2 branch?

adohe committed:
fix reuse nodePort issue

zmerlynn committed:
Merge pull request #24517 from cjcullen/automated-cherry-pick-of-#24429-upstream-release-1.2

Automated cherry pick of #24429

zmerlynn committed:
Merge pull request #24523 from jsafrane/automated-cherry-pick-of-#24035-upstream-release-1.2

Automated cherry pick of #24035

zmerlynn committed:
Merge pull request #24522 from errm/automated-cherry-pick-of-#24292-upstream-release-1.2

Automated cherry pick of #24292

zmerlynn committed:
Merge pull request #24540 from yujuhong/automated-cherry-pick-of-#24437-upstream-release-1.2

Automated cherry pick of #24437

zmerlynn committed:
Merge pull request #24543 from andyzheng0831/automated-cherry-pick-of-#24531-upstream-release-1.2

Automated cherry pick of #24531

zmerlynn committed:
Merge pull request #24566 from freehan/automated-cherry-pick-of-#22573-upstream-release-1.2

Automated cherry pick of #22573

zmerlynn committed:
Merge pull request #24581 from AdoHe/automated-cherry-pick-of-#24180-upstream-release-1.2

Automated cherry pick of #24180

zmerlynn committed:
Merge pull request #24571 from tomdee/automated-cherry-pick-of-#23985-upstream-release-1.2

Automated cherry pick of #23985

zmerlynn committed:
Merge pull request #24621 from caesarxuchao/automated-cherry-pick-of-#24006-upstream-release-1.2

Automated cherry pick of #24006

j3ffml committed:
kubectl rolling-update support for same image

Commit:
Add support for running clusters on GCI

david-mcmahon committed:
Kubernetes version v1.2.3

david-mcmahon committed:
Kubernetes version v1.2.4-beta.0

Commit:
Trusty: Add retry in curl commands

j3ffml committed:
add e2e test for same-image rolling-update

yujuhong committed:
e2e: add a dummy environment variable in the service tests

This works around the docker bug:
https://github.com/docker/docker/issues/14203

a-robinson committed:
Fix DeletingLoadBalancer event generation.

Also patch up a log statement to log the service's namespace and name
rather than potentially just logging `kind:"" apiVersion:""`.

spxtr committed:
Use zone from the test context for static IPs.

Commit:
Merge pull request #24871 from andyzheng0831/automated-cherry-pick-of-#24749-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24749

Cherry pick of #24749 on release-1.2.

Commit:
Merge pull request #24869 from yujuhong/automated-cherry-pick-of-#24830-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24830

Cherry pick of #24830 on release-1.2.

david-mcmahon committed:
Ignore all _output* directories.

mqliang committed:
fix PrepareForUpdate bug for PV and PVC

liggitt committed:
Validate deletion timestamp doesn't change on update

Commit:
Merge pull request #24936 from andyzheng0831/automated-cherry-pick-of-#24893-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24893

Cherry pick of #24893 on release-1.2.

Commit:
GCI: Update the command to get the image

Commit:
Merge pull request #24933 from liggitt/automated-cherry-pick-of-#24839-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24839

Cherry pick of #24839 on release-1.2.

Commit:
GCI: Add two GCI specific metadata pairs

wonderfly committed:
Add --image-type option to GKE kube-up

Commit:
Add an entry to the salt config to allow Debian jessie on GCE. As with the existing Wheezy image on GCE, docker is expected to already be installed in the image.

Commit:
Merge pull request #24897 from david-mcmahon/automated-cherry-pick-of-#24895-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #24895

Cherry pick of #24895 on release-1.2.

a-robinson committed:
GCE: Prefer preconfigured node tags for firewalls, if available

Commit:
Merge pull request #24999 from andyzheng0831/automated-cherry-pick-of-#24987-upstream-release-1.2

Automated cherry pick of #24987

ixdy committed:
Use --format='value()' instead of --fields with gcloud

Commit:
Merge pull request #25164 from ixdy/automated-cherry-pick-of-#25159-upstream-release-1.2

Automated cherry pick of #25159

Commit:
Merge pull request #25157 from davidopp/automated-cherry-pick-of-#23689-upstream-release-1.2

Drain pods created from ReplicaSets in 'kubectl drain'

mqliang committed:
fix PrepareForUpdate bug for HPA

Commit:
Merge pull request #25182 from a-robinson/automated-cherry-pick-of-#25148-upstream-release-1.2

Automated cherry pick of #25148 upstream release 1.2

Commit:
Merge pull request #25155 from roberthbailey/automated-cherry-pick-of-#24833-upstream-release-1.2

Automated cherry pick of #24833 upstream release 1.2

Commit:
Merge pull request #25098 from spxtr/automated-cherry-pick-of-#24827-release-1.2

Use zone from the test context for static IPs.

Commit:
Merge pull request #25152 from roberthbailey/automated-cherry-pick-of-#25123-upstream-release-1.2

Automated cherry pick of #25123

Commit:
Merge pull request #25158 from andyzheng0831/automated-cherry-pick-of-#25105-upstream-release-1.2

Automated cherry pick of #25105

Commit:
Merge pull request #25202 from roberthbailey/automated-cherry-pick-of-#24924-upstream-release-1.2

Automated cherry pick of #24924 upstream release 1.2

a-robinson committed:
Actually populate nodeTags field in GCE provider struct

Commit:
Merge pull request #25226 from a-robinson/automated-cherry-pick-of-#25225-upstream-release-1.2

Automated cherry pick of #25225 upstream release 1.2

a-robinson committed:
Don't surround node-tags list with square brackets

That's not how yaml list parsing works...

Commit:
Merge pull request #25233 from a-robinson/automated-cherry-pick-of-#25230-upstream-release-1.2

Automated cherry pick of #25230 upstream release 1.2

david-mcmahon committed:
Kubernetes version v1.2.4

david-mcmahon committed:
Kubernetes version v1.2.5-beta.0

eparis commented:
This PR has some serious problems. It has over 250 commits. What are you trying to do? Maybe I can help explain how to do it....

liggitt committed:
Return 'too old' errors from watch cache via watch stream

justinsb committed:
AWS kube-up: Increase timeout waiting for docker start

Apparently our cluster start time increased, to the point where users
are reporting spurious timeouts (#23623) and users are reporting that
increasing the timeout fixes the issue (thanks @paralin for the
suggestion and @jlfields for confirming).

Fix #23623

Commit:
GCI: Ensure that the right version of kubelet is used

Commit:
Fix hyperkube flag parsing

hyperkube kubectl cobra subcommands now work as expected.

Add unit tests with hyperkube subcommands

j3ffml committed:
Don't trim whitespace from kubectl output in e2e tests

j3ffml committed:
e2e: make ForEach fail if filter is empty, fix no-op tests

Commit:
GCI: Enable logs for upstart jobs

Commit:
Merge pull request #25598 from jlowdermilk/automated-cherry-pick-of-#25519-#25537-upstream-release-1.2

Automatic merge from submit-queue

Automated cherry pick of #25519 #25537

Cherry pick of #25519 #25537 on release-1.2. Fixes #25426.

Commit:
GCI:Trusty: Fix an issue in using 'find' commands

Commit:
GCI: Fix the condition for using the default image

Commit:
Merge pull request #25884 from maisem/automated-cherry-pick-of-#25156-upstream-release-1.2

Automated cherry pick of #25156

Commit:
Merge pull request #25450 from liggitt/automated-cherry-pick-of-#25369-upstream-release-1.2

Automated cherry pick of #25369

Commit:
Merge pull request #26088 from jlowdermilk/automated-cherry-pick-of-#24645-upstream-release-1.2

Automated cherry pick of #24645

Commit:
Fix long-standing bug in aws.stringSetToPointers

Instead of N pointers, we were returning N null pointers, followed by the real
thing. It's not clear why we didn't trip on this until now, maybe there is a
new server-side check for empty subnetID strings.

adohe committed:
fix strategy patch diff list issue

adityakali committed:
pin GCI version to milestone 52

This is mainly for pinning the 1.2 branch to GCI milestone 52
which contains correct docker and kubelet built in.
Doing this allows us to upgrade docker to v1.11 (issue #26455)
in GCI 53 without breaking the 1.2 release branch.

Commit:
Add kubectl version guard around kubectl port-forward

zmerlynn committed:
Merge pull request #26641 from andyzheng0831/automated-cherry-pick-of-#24609-#25504-#25763-#25609-#25706-upstream-release-1.2

Automated cherry pick of #24609 #25504 #25763 #25609 #25706

Commit:
Trusty: fix 'find' commands and add k8s license and motd info

janetkuo committed:
Avoid comparing pod-template-hash when getting a deployment's new/old RSes

janetkuo committed:
Fix nil pointer dereference in deployment e2e test

Commit:
Merge branch 'release-1.2' into automated-cherry-pick-of-#25405-upstream-release-1.2

Commit:
Merge pull request #26826 from adityakali/automated-cherry-pick-of-#26582-kube-release-1.2

Automated cherry pick of #26582

Commit:
Merge branch 'release-1.2' into automated-cherry-pick-of-#25405-upstream-release-1.2

Commit:
Merge pull request #26918 from ihmccreery/automated-cherry-pick-of-#23445-upstream-release-1.2

Automated cherry pick of #23445

Commit:
Trusty: fix the 'ping' issue and fluentd-gcp issue #26379

Commit:
Merge pull request #27202 from andyzheng0831/automated-cherry-pick-of-#27136-upstream-release-1.2

Trusty: fix the 'ping' issue and fluentd-gcp issue #26379

janetkuo committed:
Merge branch 'release-1.2' into automated-cherry-pick-of-#26792-upstream-release-1.2

Commit:
Merge pull request #26922 from janetkuo/automated-cherry-pick-of-#26792-upstream-release-1.2

Automated cherry pick of #26792 upstream release 1.2

Commit:
Merge pull request #27288 from andyzheng0831/automated-cherry-pick-of-#26741-upstream-release-1.2

Automated cherry pick of #26741

Commit:
in nsenter mounter, read hosts PID 1 /proc/mounts to list the mounts

Signed-off-by: Huamin Chen <hchen@redhat.com>

Commit:
Merge pull request #27448 from ihmccreery/automated-cherry-pick-of-#27310-upstream-release-1.2

Automated cherry pick of #27310

colhom committed:
Update spf13/cobra dependency to match master

janetkuo committed:
Retry Pod/RC updates in kubectl rolling-update

Commit:
Merge pull request #26754 from colhom/automated-cherry-pick-of-#25512-upstream-release-1.2

Automated cherry pick of #25512

Commit:
Trusty: Fix the motd reset

Commit:
Merge branch 'release-1.2' into automated-cherry-pick-of-#25405-upstream-release-1.2

Commit:
Fix hack/update-api-reference-docs.sh

- update toplevel api object pattern in swagger doc script
- make hack/update-api-reference-docs.sh indepenent from master

Commit:
Merge pull request #26098 from roberthbailey/automated-cherry-pick-of-#25405-upstream-release-1.2

AWS kube-up: Increase timeout waiting for docker start

Commit:
Merge pull request #27664 from Clarifai/automated-cherry-pick-of-#26331-upstream-release-1.2

Fix long-standing bug in aws.stringSetToPointers

Commit:
Merge pull request #27641 from andyzheng0831/trusty-fix

Trusty: Fix the motd reset

zmerlynn committed:
GCE provider: Limit Filter calls to regexps rather than insane blobs

Filters can't exceed 4k, and GET requests against the GCE API are also
limited, so these break down in different ways at different cluster
counts. Fix it by introducing an advisory node-instance-prefix
configuration in the GCE provider that can hint the
EnsureLoadBalancer/UpdateLoadBalancer code (and the firewall
creation/update code). If it's not there, or wrong (a hostname that's
registered violates it), just ignore it and grab the whole project.

wojtek-t committed:
Write cluster name in prepare-upgrade

Commit:
Merge pull request #27113 from AdoHe/automated-cherry-pick-of-#26418-upstream-release-1.2

Automated cherry pick of #26418

Commit:
Merge pull request #27799 from zmerlynn/automated-cherry-pick-of-#27741-upstream-release-1.2

Automated cherry pick of #27741

bprashanth committed:
Bump glbc version

Commit:
Merge pull request #27817 from caesarxuchao/automated-cherry-pick-of-#27706-upstream-release-1.2

Automated cherry pick of #27706

zmerlynn committed:
GCE provider: Create TargetPool with 200 instances, then update with rest

Tested with 2000 nodes, this actually meets the GCE API specifications
(which is nutty). Previous PR (#25178) was based on a mistaken
understanding of a poorly documented set of limitations, and even
poorer testing, for which I am embarassed.

Commit:
Merge pull request #27865 from zmerlynn/manual-cherrypick-25178

Manual cherry pick of #27829 on release-1.2

Commit:
Merge pull request #27861 from zmerlynn/automated-cherry-pick-of-#27814-upstream-release-1.2

Automated cherry pick of #27814

Commit:
Merge pull request #28018 from rootfs/automated-cherry-pick-of-#27380-upstream-release-1.2

Automated cherry pick of #27380

Commit:
Revert "Automated cherry pick of #27814"

Commit:
Merge pull request #28050 from kubernetes/revert-27861-automated-cherry-pick-of-#27814-upstream-release-1.2

Revert "Automated cherry pick of #27814"

Commit:
Merge pull request #28053 from jlowdermilk/automated-cherry-pick-of-#27509-upstream-release-1.2

Automated cherry pick of #27509

david-mcmahon committed:
Kubernetes version v1.2.5

david-mcmahon committed:
Kubernetes version v1.2.6-beta.0

Commit:
Merge pull request #28163 from piosz/cherry-pick-of-#27763-upstream-release-1.2

Cherry pick of #27763: Write cluster name in prepare-upgrade

roberthbailey commented:
@kgelinas it looks like you are trying to pull commits from 1.2 into the master branch. Can you create a new pull request with just your changes against the master branch?
